Lib/test/test_readline.py

```diff
@@ -210,13 +212,49 @@ def display(substitution, matches, longest_match_length):
         self.assertIn(b"result " + expected + b"\r\n", output)
         self.assertIn(b"history " + expected + b"\r\n", output)
 
+    def test_history_size(self):
+        history_size = 10
+        temp_dir = tempfile.mkdtemp()
+        try:
+            inputrc = os.path.join(temp_dir, "inputrc")
+            with io.open(inputrc, "wb") as f:
+                f.write(b"set history-size %d\n" % history_size)
+            env = os.environ.copy()
+            env["INPUTRC"] = inputrc
+
+            history_file = os.path.join(temp_dir, "history")
+            with io.open(history_file, "wb") as f:
+                # history_size * 2 items used to crash readline
+                data = b"".join(b"item %d\n" % i
+                                for i in range(history_size * 2))
+                f.write(data)
+
+            script = """
+import readline
+import sys
+
+history_file = "{}"
+readline.read_history_file(history_file)
+input()
+readline.write_history_file(history_file)
+""".format(history_file)
+
+            run_pty(script, input=b"last input\r", env=env)
+
+            with io.open(history_file, "rb") as f:
+                lines = f.readlines()
+            self.assertEqual(len(lines), history_size)
+            self.assertEqual(lines[-1].strip(), b"last input")
+        finally:
+            shutil.rmtree(temp_dir)
```

Comment:
Can't we use `self.addCleanup(shutil.rmtree, temp_dir)` instead of this try..finally block?